pycqed/instrument_drivers/meta_instrument/heterodyne.py

```diff
@@ -23,75 +23,295 @@ class HeterodyneInstrument(Instrument):
                            set_cmd=self.do_set_RF_power,
                            get_cmd=self.do_get_RF_power)
-        self.add_parameter('single_sideband_demod',
-                           label='Single sideband demodulation',
-                           parameter_class=ManualParameter)
-        self.add_parameter('acquisition_instr',
-                           set_cmd=self._do_set_acquisition_instr,
-                           get_cmd=self._do_get_acquisition_instr,
-                           vals=vals.Anything())
         self.add_parameter('acquisition_instr_controller',
-                           set_cmd=self._do_set_acquisition_instr_controller,
-                           get_cmd=self._do_get_acquisition_instr_controller,
+                           set_cmd=self.do_set_acquisition_instr_controller,
+                           get_cmd=self.do_get_acquisition_instr_controller,
                            vals=vals.Anything())
-        self.add_parameter('nr_averages',
+        self.acquisition_instr_controller(acquisition_instr_controller)
+        self._RF_power = None
+
+    def common_init(self, name, LO, AWG, acquisition_instr='CBox',
+                    single_sideband_demod=False, **kw):
+        logging.info(__name__ + ' : Initializing instrument')
+        Instrument.__init__(self, name, **kw)
+
+        self.LO = LO
+        self.AWG = AWG
+        self.add_parameter('frequency', label='Heterodyne frequency',
+                           units='Hz', vals=vals.Numbers(9e3, 40e9),
+                           get_cmd=self.do_get_frequency,
+                           set_cmd=self.do_set_frequency)
+        self.add_parameter('f_RO_mod', label='Intermodulation frequency',
+                           units='Hz', vals=vals.Numbers(-600e6, 600e6),
+                           set_cmd=self.do_set_f_RO_mod,
+                           get_cmd=self.do_get_f_RO_mod)
+        self.add_parameter('single_sideband_demod', vals=vals.Bool(),
+                           label='Single sideband demodulation',
+                           parameter_class=ManualParameter,
+                           initial_value=single_sideband_demod)
+        self.add_parameter('acquisition_instr', vals=vals.Strings(),
+                           label='Acquisition instrument',
+                           set_cmd=self.do_set_acquisition_instr,
+                           get_cmd=self.do_get_acquisition_instr)
+        self.add_parameter('nr_averages', label='Number of averages',
+                           vals=vals.Numbers(min_value=0, max_value=1e6),
                            parameter_class=ManualParameter,
-                           initial_value=1024,
-                           vals=vals.Numbers(min_value=0, max_value=int(1e6)))
+                           initial_value=1024)
+        self.add_parameter('status', vals=vals.Enum('On','Off'),
+                           set_cmd=self.do_set_status,
```

Comment:
Should also be private. 

On a separate note, all these self.do_set_status (and a lot of the other ones) do is flag that some parameters have changed. This is the perfect use case for a configuration parameter : https://github.com/DiCarloLab-Delft/PycQED_py3/blob/master/pycqed/instrument_drivers/meta_instrument/kernel_object.py#L243. 

But don't worry about changing that now, I still have to put that in a PR to QCoDeS